Do not use root user inside container

Godin opened the pull request:
On start of container create dedicated "travis" user with UID and GID
specified by environment variables DEV_UID and DEV_GID respectively,
and step-down from root user before execution of command provided by
user.

Demo:

```
$ ls -lA test
ls: cannot access test: No such file or directory

$ docker run -it -e DEV_UID=$UID -e DEV_GID=$GID -v $PWD:/home/travis --rm local-travis \
id && touch test
uid=1000(travis) gid=1000(travis) groups=1000(travis)

$ ls -lA test 
-rw-r--r-- 1 godin godin 0 Sep 13 22:08 test
```

Note that in case if user UID conflicts with UID already existing in container, message will look like:

```
$ docker run -it -e DEV_UID=0 -e DEV_GID=0 -v $PWD:/home/travis --rm local-travis bash
groupadd: GID '0' already exists
```

The Linux Standard Base Core Specification specifies that range of UIDs from 0 to 500 is reserved. Leopard Security Config p.60 specifies that "new users created using the Accounts pane of System Preferences are assigned user IDs starting at 501". Thus IMO appearance of conflict is unlikely:

```
$ docker run -it -e DEV_UID=$UID -e DEV_GID=$GID -v $PWD:/home/travis --rm local-travis \
awk -F: '{if ($3 >= 500) { print $1 ":" $3 } }' /etc/passwd
nobody:65534
travis:1000
```

However, if this ever happens, then we can add "--non-unique" option for "useradd" and "groupadd".

dgageot commented:
@Godin I have no `$GID` env. variable on OSX. We could replace `$UID` with `$(id -u)` and `$GID` with `$(id -g)`. WDYT?

Godin committed:
Do not use root user inside container

On start of container create dedicated "travis" user with UID and GID
specified by environment variables DEV_UID and DEV_GID respectively,
and step-down from root user before execution of command provided by
user.

Godin commented:
@dgageot looks like `$GID` provided by zsh even on Linux, so I made a change as you suggested (PR updated with force-push)
